### PR TITLE
Pv2 410 infrastructure component   funding source   act2 incentives

### PR DESCRIPTION
**Pull request by abroodie**

### CODE REVIEW

File: src/SFA.DAS.Payments.FundingSource.NonLevyFundedService/Handlers/IncentiveRequiredPaymentEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            IIncentiveRequiredPaymentProcessor incentiveProcessor)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.executionContext = executionContext ?? throw new ArgumentNullException(nameof(executionContext));
+            this.incentiveProcessor = incentiveProcessor ?? throw new ArgumentNullException(nameof(incentiveProcessor));
+        }
+
+        public async Task Handle(IncentiveRequiredPayment message, IMessageHandlerContext context)
+        {
+            try
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.FundingSource.NonLevyFundedService/Handlers/IncentiveRequiredPaymentEventHandler.cs]: You need to specify --  Jobid --  like var 
currentExecutionContext = (ESFA.DC.Logging.ExecutionContext)executionContext;
            currentExecutionContext.JobId = message.JobId.ToString();

Comment (abroodie) [reply src/SFA.DAS.Payments.FundingSource.NonLevyFundedService/Handlers/IncentiveRequiredPaymentEventHandler.cs]: Done

File: src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using SFA.DAS.Payments.Model.Core.Entities;
+
+namespace SFA.DAS.Payments.AcceptanceTests.Core.Data
+{
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs]: I've made it static

File: src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public class Helper
+    {
+        public static TransactionType GetTransactionType(string transactionType)
+        {
+            transactionType = transactionType.ToLower();
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs]: ideally yes, but the feeding specs use Learning (TT1) etc so we can't get away from this at the moment

File: src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                : transactionType.Contains("completion")
+                    ? Model.Core.Entities.TransactionType.Completion
+                    : transactionType.Contains("balancing")
+                        ? Model.Core.Entities.TransactionType.Balancing
+                        : transactionType.Contains("first16to18employerincentive")
+                            ? Model.Core.Entities.TransactionType.First16To18EmployerIncentive
+                            : throw new InvalidOperationException($"Unknown transaction type: '{transactionType}'");
+        }
+
+        public static FundingSourceType GetFundingSourceType(string fundingSourceType)
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs]: changed

File: src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using SFA.DAS.Payments.Model.Core.Entities;
+
+namespace SFA.DAS.Payments.AcceptanceTests.Core.Data
+{
+    public class Helper
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/Data/Helper.cs]: I've called it EnumHelper

**Pull request merged**